pdns/libssl.cc

```diff
@@ -702,12 +756,21 @@ std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)> libssl_init_server_context(const TLS
     SSL_CTX_sess_set_cache_size(ctx.get(), config.d_maxStoredSessions);
   }
 
+  long mode = 0;
 #ifdef SSL_MODE_RELEASE_BUFFERS
   if (config.d_releaseBuffers) {
-    SSL_CTX_set_mode(ctx.get(), SSL_MODE_RELEASE_BUFFERS);
+    mode |= SSL_MODE_RELEASE_BUFFERS;
+  }
+#endif
+
+#ifdef SSL_MODE_ASYNC
+  if (config.d_asyncMode) {
+    mode |= SSL_MODE_ASYNC;
   }
 #endif
```

Comment:
Add an #else part to make setting `asynMode` an error/warning if not available?
Also a question: does it make sense to set `asyncMode` if no engine is set? If not, it would be nice to detect and warn about that.

Reply:
Sure, I will add the warning.
As for your second question, it does not make sense today, but I'm not sure that will not change in the future. I believe it would make sense for OpenSSL to offer more asynchronous processing mechanisms.